### DISCUSSION TITLE
tutorials/voyage-part1/11-entities-and-the-query-plan

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#63] [course feedback]
# tutorials/voyage-part1/11-entities-and-the-query-plan

https://www.apollographql.com/tutorials/voyage-part1/11-entities-and-the-query-plan

**Comment by edwardyun12** [comment]
If you query the other way where you query the location.name first and then query the latestreviews. Would it trigger the error as you don't have a entity representation?

**Reply by mabuyo** [reply]
Hey @edwardyun12 ! Can you clarify what you mean? What would the query look like in your example?

**Reply by torressam333** [reply]
If you do something like this you mean:

```
query Query($locationId: ID!) {
  location(id: $locationId) {
    name
  }
  latestReviews {
    comment
  }
}
```

?

If so, this resolves just fine

```
{
  "data": {
    "location": {
      "name": "The Living Ocean of New Lemuria"
    },
    "latestReviews": [
      {
        "comment": "This is simply unbelievable! It's the perfect solution for our business. Really good. I don't always clop, but when I do, it's because of planet"
      },
      {
        "comment": "Planet is exactly what our business has been ....
```

**Comment by rtpa25** [comment]
Order doesn't really matter, as the router makes a query plan after reading through all the fields you requested for, then it makes any kinds of requests to other subgraphs, hence order of field does not matter as during time of sending a request --> router has already registered all the fields it needs to query from different subgraphs, in it's query plan

**Comment by Dolvur** [comment]
Looking at the pictures, and following the reasoning, I get how you can connect the entities together. However, how is it possible in the given example to query:
```graphql
query ExampleQuery {
  latestReviews {
    id
    comment
    rating
    location {
        name
    }
  }
}
```
When the return type of `latestReviews` is `[Review!]!` and type `Review` does not contain any `location` field

```graphql
type Review {
  id: ID!
  "Written text"
  comment: String
  "A number from 1 - 5 with 1 being lowest and 5 being highest"
  rating: Int
}
```
So, even though we have defined a `Location` entity/stub, we don't mention that it is possible to return Location from the given query, so the query in the example should not be possible right? If we add `location: Location` as a field to `Review` then it would work right?

**Reply by Dolvur** [reply]
Nvm, this was done in the next lesson (location: Location is added to Review). I stopped just short of the next lesson and was wondering if there was some hidden connection 😅
